### ISSUE TITLE
Feature request: path groups

### ISSUE THREAD

**Issue by robobenklein** [aplbrain/grand-cypher#26]
For example, the syntax:

```
MATCH p=(n1 {type: "compilation_unit"})-[]->(n2 {type: "class_declaration"})-[*2]->(n3 {type: "method_declaration"})-->(n4 {text: "Main", type:"identifier"})
RETURN p
```

The return type for a path from neo4j has `start, end, segments`, and notably includes nodes or edge that would be traversed by the edge `*` operator.

I can still work around it, but I would very much like to have access to variable-length paths!

**Comment by j6k4m8** [comment]
Hey @robobenklein! Thanks for the comment, and yes this sounds like a great feature addition! Adding to our backlog :)

**Comment by davidmezzetti** [comment]
@j6k4m8 - I believe this can be closed as fixed with #37?

**Comment by j6k4m8** [comment]
Yes confirmed! Thank you @davidmezzetti !!
